src/riot-rs-threads/src/lib.rs

```diff
@@ -108,9 +108,16 @@ impl Threads {
         }
     }
 
-    // fn get_unchecked(&self, thread_id: ThreadId) -> &Thread {
-    //     &self.threads[thread_id as usize]
-    // }
+    /// Returns access to any thread data.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `thread_id` is >= [`THREADS_NUMOF`].
+    /// If the thread for this `thread_id` is in an invalid state, the
+    /// data in the returned [`Thread`] is undefined, i.e. empty or outdated.
+    fn get_unchecked(&self, thread_id: ThreadId) -> &Thread {
```

Comment:
should this be unsafe? (is the std lib `get_unchecked()` usually unsafe?)

Reply:
It's an non-mutable version of the existing `get_unchecked_mut` function. 
I'm, not sure if it has to be made unsafe because we explicitly documented that it can panic. `core::ops::Index::index` is also not unsafe. 
On the other hand, it could cause undefined behavior if the thread_id is not out-of-bound, but the thread in an invalid state.
I guess the correct way would be to make it `unsafe` and then bubble up the unsafe to the user (e.g. so that `thread_flags::set` is also unsafe, since this internally calls `get_unchecked_mut`)  and/ or add checks everywhere that the thread id is valid. But I assume that would impact performance quite a bit.

Reply:
Ok, so get_unchecked(...) panics if thread_id is out-of-bounds. That doesn't make it unsafe.
The thread object reference is always valid memory, so no unsafe there, either.
The object might be in a "weird" state, but that doesn't make it unsafe, either. I don't see how this could lead to "Undefined Behavior". It might lead to a thread in Invalid state to be added to a runqueue, but that's not "Undefined Behavior".

=> IMO, no unsafe necessary.

Down the line, we might want to explore if it is worth the effort to introduce a "ValidThreadId" type to make API misuse less possible, or if we can contain our reasoning to internals.